CMakeLists.txt

```diff
@@ -20,30 +23,74 @@ include(${_project_options_SOURCE_DIR}/Index.cmake)
 # Set the project name and language
 project(myproject LANGUAGES CXX)
 
+if(MSVC)
+  # Make sure that all supported configuration types have their
+  # associated conan packages available. You can reduce this
+  # list to only the configuration types you use
+  set(CMAKE_CONFIGURATION_TYPES
+      Debug
+      Release
+      RelWithDebInfo
+      MinSizeRel)
+endif()
+
+# Set defaults for options enabled with the project_options project
+# and the GlobalOptions utility below
+#
+# Note: These defaults represent those suggested by Jason Turner in
+# C++ Best Practices
+set(ENABLE_CPPCHECK
+    ON
+    CACHE BOOL "Enable CppCheck analysis")
+set(ENABLE_CLANG_TIDY
+    ON
+    CACHE BOOL "Enable clang-tidy analysis")
+set(ENABLE_CACHE
+    ON
+    CACHE BOOL "Enable ccache")
+set(ENABLE_COVERAGE
+    ON
+    CACHE BOOL "Enable coverage testing")
+set(ENABLE_CONAN
+    ON
+    CACHE BOOL "Enable conan for dependencies")
+set(ENABLE_SANITIZER_ADDRESS
+    ON
+    CACHE BOOL "Enable address sanitizer for runtime memory checking")
+set(ENABLE_SANITIZER_UNDEFINED_BEHAVIOR
+    ON
+    CACHE BOOL "Enable undefined behavior sanitizer for runtime checking")
+set(WARNINGS_AS_ERRORS
+    ON
+    CACHE BOOL "Enable undefined behavior sanitizer for runtime checking")
```

Comment:
Why add all these global options while you can set the defaults by uncommenting the function arguments?